src/snapred/backend/recipe/algorithm/SmoothDataExcludingPeaksAlgo.py

```diff
@@ -118,6 +118,14 @@ def PyExec(self):
         weight_ws = self.mantidSnapper.mtd[self.weightWorkspaceName]
 
         numSpec = weight_ws.getNumberHistograms()
+
+        allZeros = all(np.all(weight_ws.readX(index) == 0) for index in range(numSpec))
```

Comment:
Because weight_ws.readX(index) is used to check if all the elements within each numpy array are zero. Then built in all function is used to verify that the condition holds true for all generate arrays ensuring every element in every array is zero.